db/migrate/20210614145845_add_email_name_id_format_to_service_providers.rb

```diff
@@ -0,0 +1,17 @@
+class AddEmailNameIdFormatToServiceProviders < ActiveRecord::Migration[6.1]
+  def change
+    add_column :service_providers, :allow_email_nameid_format, :boolean, default: false
```

Comment:
`strong_migrations` is flagging this in CI for PG 9.6:

```
=== Dangerous operation detected #strong_migrations ===

Adding a column with a non-null default blocks reads and writes while the entire table is rewritten.
Instead, add the column without a default value, then change the default.

class AddEmailNameIdFormatToServiceProviders < ActiveRecord::Migration[6.1]
  def up
    add_column :service_providers, :allow_email_nameid_format, :boolean
    change_column_default :service_providers, :allow_email_nameid_format, false
  end

  def down
    remove_column :service_providers, :allow_email_nameid_format
  end
end

Then backfill the existing rows in the Rails console or a separate migration with disable_ddl_transaction!.

class BackfillAddEmailNameIdFormatToServiceProviders < ActiveRecord::Migration[6.1]
  disable_ddl_transaction!

  def up
    ServiceProvider.unscoped.in_batches do |relation| 
      relation.update_all allow_email_nameid_format: false
      sleep(0.01)
    end
  end
end
```

Correct me if I'm wrong, but if the column is `nil` for existing SP's that's not really a big deal, right? It will evaluate to `false` wherever we check for the setting, so I think I can just skip the second migration in their example.

Reply:
removing `default: false` so it defaults to `nil` is probably safe

Reply:
thanks!